sample_apps/simple_app_python/example_app.py

```diff
@@ -187,14 +215,157 @@ def do_main(args) -> bool:
+
+    # Server needs to authenticate the client; hence 'Purpose.CLIENT_AUTH'
+    context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+    context.verify_mode = ssl.CERT_REQUIRED
+
+    cert_outfilename = data_dir + '/' + SERVER_ROOT_SIGNED_CERT
+    key_outfilename  = data_dir + '/' + SERVER_ROOT_SIGNED_KEYF
+    context.load_cert_chain(certfile = cert_outfilename, keyfile = key_outfilename)
+
+    ca_root_outfilename = data_dir + '/' + ROOT_POLICY_CERT
+    context.load_verify_locations(cafile = ca_root_outfilename)
+
+    bindsocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    bindsocket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    bindsocket.bind((server_app_host, server_app_port))
+    bindsocket.listen(10)
+    print('\n', fnl(), 'Waiting for client ...')
+
+    new_socket, fromaddr = bindsocket.accept()
+    print('\n', fnl(), 'Client connected: ', fromaddr[0], ":", fromaddr[1])
+
+    secure_sock = context.wrap_socket(new_socket, server_side=True)
+
+    try:
+        data = secure_sock.recv(1024)
+        print('\n', fnl(), 'Received from client: ', str(data, 'UTF-8'))
+
+        ret_hdr = 'Return back to client: '
+        print('\n', fnl(), 'Return message back to client:', ret_hdr + str(data, 'UTF-8'))
+        secure_sock.write(bytes(ret_hdr, 'UTF-8') +  data)
+    finally:
+        secure_sock.close()
+        bindsocket.close()
+
+    return True
+
+###############################################################################
+def client_dispatch(data_dir, server_app_host, server_app_port):
+    """
+    Start a client app that sends a message via secure SSL connection.
+    """
+    print(fnl(), 'Start client application process ...\n')
+
+    # Client needs to authenticate the server; hence 'Purpose.SERVER_AUTH'
+    context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+    context.verify_mode = ssl.CERT_REQUIRED
+
+    cert_outfilename = data_dir + '/' + CLIENT_ROOT_SIGNED_CERT
+    key_outfilename  = data_dir + '/' + CLIENT_ROOT_SIGNED_KEYF
+    context.load_cert_chain(certfile = cert_outfilename, keyfile = key_outfilename)
+
+    ca_root_outfilename = data_dir + '/' + ROOT_POLICY_CERT
+    context.load_verify_locations(cafile = ca_root_outfilename)
+
+    bindsocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    bindsocket.setblocking(1)
+    bindsocket.connect((server_app_host, server_app_port))
+
+    secure_sock = context.wrap_socket(bindsocket, server_side=False,
+                                      server_hostname=server_app_host)
+
+    send_msg = 'hello'
+    try:
+        secure_sock.send(bytes(send_msg, 'UTF-8'))
+
+        recv_data = secure_sock.recv(1024)
+        recv_data_str = str(recv_data, 'UTF-8')
+
+        print('\n', fnl(), 'Received message from server:', recv_data_str)
+
+        # Server should have prepended this to our message and returned it
+        assert recv_data_str == 'Return back to client: ' + send_msg
```

Comment:
We also do a bit of cross-checking in this app, for the expected return-message.